server/lib/payments.js

```diff
@@ -613,7 +613,11 @@ const sendOrderConfirmedEmail = async (order, transaction) => {
   const user = await order.getUserForActivity();
   const host = await collective.getHostCollective();
   const parentCollective = await collective.getParentCollective();
-  const customMessage = collective.settings?.customEmailMessage || parentCollective?.settings?.customEmailMessage;
+  const customMessage =
+    collective.settings?.customEmailMessage ||
+    parentCollective?.settings?.customEmailMessage ||
+    collective.settings?.customEmailMessage?.thankYou ||
+    parentCollective?.settings?.customEmailMessage?.thankYou;
```

Comment:
So initially we had, 

```
const customMessage = collective.settings?.customEmailMessage || parentCollective?.settings?.customEmailMessage;
```
But once the front-end PR, https://github.com/opencollective/opencollective-frontend/pull/8044 is merged we will store custom email message templates in, `customEmailMessage?.thankYou`. Hence the two new fields, `collective.settings?.customEmailMessage?.thankYou || parentCollective?.settings?.customEmailMessage?.thankYou` added to the above condition. 

But after the front-end PR is merged we no longer need, `collective.settings?.customEmailMessage || parentCollective?.settings?.customEmailMessage`  part in this condition as everything thereafter is stored in `settings?.customEmailMessage.thankYou`. Hence we delete these old fields in https://github.com/opencollective/opencollective-api/pull/7784

Reply:
Here's my understanding of how this condition is going to act:

**When using the legacy settings**

```es6
settings = { customEmailMessage: 'Hello World' }
```

`customMessage` will be set to 'Hello World'. So far so good.

**When using the new settings**

```es6
settings = { customEmailMessage: { thankYou: 'Hello World' } }
```

`collective.settings?.customEmailMessage` is not nullish, so `customMessage` will be set to `{ thankYou: 'Hello World' }`. How is the email template going to react if it receives an object rather than a string for `customMessage`?

Reply:
@Betree : Ops you are correct, this condition has a problem, we need to change the order of things so that it first checks for the `settings.customEmailMessage.thankYou` instead of `settings.customEmailMessage`. Thanks for pointing it out. I've corrected it.